sys/kern/Cargo.toml

```diff
@@ -33,6 +33,7 @@ phash-gen = {path = "../../build/phash-gen"}
 
 [features]
 vectored-interrupts = []
+riscv-supervisor-mode = []
```

Comment:
This seems okay for now, but I don't know if there is a better way to organize the feature flags to limit this to riscv arch only.

Reply:
Agreed. While deeply preferable, I'm not sure on how to achieve this today, and believe we should aim to address it in a follow-up

Reply:
Ack

Reply:
Maybe something like this in the kernel `build.rs`?  
```
fn verify_cfg() -> Result<(), ()> {
    target = env::var("TARGET").unwrap();
    cfg_if::cfg_if! {
        if #[cfg(feature = "riscv-supervisor-mode")] && !target.starts_with("riscv") {
            compile_error!();
        }
    }
}
```
I think there is a cfg for the target too...

Reply:
Yeah, I think that would be a good check to have in place.